### ISSUE TITLE
Unable to find any doc to run local devnet using Erigon

### ISSUE THREAD

**Issue by Uttam-Singhh** [ledgerwatch/erigon#2588]
I want to run Erigon on my local devnet but I am unable to find any documents or any steps to do so??
I am a beginner please let me know if there is any document or article.

Here is an example of running local devnet using Geth - https://medium.com/datawallet-blog/how-to-deploy-a-local-private-ethereum-blockchain-c2b497f068f4
I really want any article like this for Erigon too.

Also, the discord link to join the server is expired, Please send me another invite on - uttamkhanduja@yahoo.in if possible.

I really want to explore erigon!

**Comment by Uttam-Singhh** [comment]
So there is no other way to start a stand-alone chain?? @AskAlexSharov @mmgen

**Comment by mmgen** [comment]
> So there is no other way to start a stand-alone chain?? @AskAlexSharov @mmgen

As far as I know, no.

**Comment by Uttam-Singhh** [comment]
@AskAlexSharov I'd love to work on this functionality to init a custom chain with its own genesis or bring back the functionality if it was already there!
Should I create a new issue??
and also I would be really grateful if someone can guide me more on this!

**Comment by AskAlexSharov** [comment]
Sure. 

I advise you start from re-introducing `geth init` to setup custom genesis block. 
We have general ticket: https://github.com/ledgerwatch/erigon/issues/2030 (if you don't need `geth import` it's fine).
You can get source code of `geth import` - I expect that amount of required changes will not be very big. 
Feel free to do it in multiple PR's - do as comfortable or as make sense. 

We have only 1 non-obvious requirement:
- we don't wan't have Account Manager - don't want store private key in Erigon's memory or generate them. Because we believe it will simplify for us Security Audit in future.

You can ask questions in Discord, me or somebody online will answer. 

I will start from another side - I will check is it easy to mine blocks for dev-net or not. Do you need dev-net based on Ethash or Clique is also fine?

**Comment by Uttam-Singhh** [comment]
> Sure.
> 
> I advise you start from re-introducing `geth init` to setup custom genesis block.
> We have general ticket: #2030 (if you don't need `geth import` it's fine).
> You can get source code of `geth import` - I expect that amount of required changes will not be very big.
> Feel free to do it in multiple PR's - do as comfortable or as make sense.
> 
> We have only 1 non-obvious requirement:
> 
> * we don't wan't have Account Manager - don't want store private key in Erigon's memory or generate them. Because we believe it will simplify for us Security Audit in future.
> 
> You can ask questions in Discord, me or somebody online will answer.
> 
> I will start from another side - I will check is it easy to mine blocks for dev-net or not. Do you need dev-net based on Ethash or Clique is also fine?

Thanks for your reply!
I don't need `geth import`. I just want to run a stand alone chain(local devnet).
Also as you advised, how can I start implementing  `geth init`, what are the changes that has to done?
Do I need to do changes in Config.go file under params ? please provide a reference if possible!

**Comment by mmgen** [comment]
> We have only 1 non-obvious requirement:
> 
>     * we don't wan't have Account Manager - don't want store private key in Erigon's memory or generate them. Because we believe it will simplify for us Security Audit in future.

Parity, for example, has a single hard-coded faucet address and private key for devnet. Would such an approach be acceptable?

**Comment by Uttam-Singhh** [comment]
> > Do you need dev-net based on Ethash or Clique is also fine?
> 
> I can't speak for @Uttam-Singhh, but the Clique engine (mines blocks automatically whenever a tx appears in the mempool) would be fine for me. I'm not interested in mining.

Yeah Clique works for me too!

**Comment by AskAlexSharov** [comment]
`what are the changes that has to done?` - I don't know - but I'm sure that part of code is not supper different from geth.

**Comment by Uttam-Singhh** [comment]
> `what are the changes that has to done?` - I don't know - but I'm sure that part of code is not supper different from geth.

oh okay!😶
@mmgen Do you have any Idea??

**Comment by mmgen** [comment]
> @mmgen Do you have any Idea??

I'm not that familiar with the code, nor am I a Go programmer, but for starters you might look at `DeveloperGenesisBlock()` in `core/genesis.go`.

**Comment by AskAlexSharov** [comment]
About next steps in article https://medium.com/datawallet-blog/how-to-deploy-a-local-private-ethereum-blockchain-c2b497f068f4 
4. i think you don't need `geth account new` command: create private/public key pair by any other utils. 
For example: https://kobl.one/blog/create-full-ethereum-keypair-and-address/
Or by geth. Or by Metamask. Or any other way. 
Erigon has code to create address: 
```
privateKeyECDSA :=ecdsa.GenerateKey(S256(), rand.Reader)
address := crypto.PubkeyToAddress(privateKeyECDSA.PublicKey)
``` 
but we will not add command to generate private keys for users - by security reasons. 
7. When Erigon node starts it prints `enode` id: 
`Started P2P networking                   version=65 self=enode://7d461b20dda15886c17bf30af7bf101ad94db0567d87c9a357bc159a3c765d4f54435fe48e1d695518e7e41068c7dcb1f274c5bc594ad3c0b0b8c48ce3de078e@127.0.0.1:30304 name=erigon/v2021.09.2/darwin-amd64/go1.17`
collect all of them and pass to --staticpeers cli flag
If you like - you can copy from geth functionality which allow storing static peers in .toml file

8. Only way to send transactions by Erigon is to use RPC method eth_sendRawTransaction 
You will need find some way to create raw transaction. 

9. We have cli flags to start miner and set etherbase. 

So, just go ahead step by step.

**Comment by Uttam-Singhh** [comment]
Thank you @AskAlexSharov for guidance!


> 4. i think you don't need `geth account new` command: create private/public key pair by any other utils.
> For example: https://kobl.one/blog/create-full-ethereum-keypair-and-address/
> Or by geth. Or by Metamask. Or any other way.

I will use Metamask for this but what about nodes?
I will have to create local node or not?


> 7. When Erigon node starts it prints `enode` id:
> `Started P2P networking version=65 self=enode://7d461b20dda15886c17bf30af7bf101ad94db0567d87c9a357bc159a3c765d4f54435fe48e1d695518e7e41068c7dcb1f274c5bc594ad3c0b0b8c48ce3de078e@127.0.0.1:30304 name=erigon/v2021.09.2/darwin-amd64/go1.17`
> collect all of them and pass to --staticpeers cli flag

Are you talking about running the main chain?
I need to collect all of them from the running main chain or any other network?

**Comment by AskAlexSharov** [comment]
“ I will have to create local nodes or not?” - yes, same as in article. 

“ Are you talking about running the main chain?” - no, you will create some nodes for your chain.

**Comment by mmgen** [comment]
> but we will not add command to generate private keys for users - by security reasons.

How about using a hard-coded private key/address for the faucet, as OpenEthereum does? Then you could just start up in devmode with a single command-line flag, say `--chain=dev`, with no setup required.

**Comment by AskAlexSharov** [comment]
Ah, yes, need to add

**Comment by Uttam-Singhh** [comment]
@AskAlexSharov I am stuck please help!
I am able to run the devnet, i also connected it using metamask, I tried to do a transaction it showed the transaction  was submitted 👇

<img width="1440" alt="Screenshot 2021-09-07 at 12 51 40 PM" src="https://user-images.githubusercontent.com/63050765/132304364-1d30ab82-d2ac-4e93-9b77-504582b4d1b3.png">


But the blocks are still not increasing, they are stuck...🤷‍♂️

**Comment by AskAlexSharov** [comment]
Show me logs of Erigon

**Comment by Uttam-Singhh** [comment]
Here it is -   The transaction was fired at `12.48.30`

<img width="1440" alt="Screenshot 2021-09-07 at 1 25 48 PM" src="https://user-images.githubusercontent.com/63050765/132306839-55d6ebdb-abdb-4469-9a6c-c5f6b82f5d5b.png">

**Comment by AskAlexSharov** [comment]
And cli command to run erigon?

**Comment by Uttam-Singhh** [comment]
`./build/bin/erigon --datadir dev --chain dev --mine --miner.sigkey=`

**Comment by AskAlexSharov** [comment]
I think `--miner.sigkey` flag was introduced by mistake. 
Let me take a look.

**Comment by Uttam-Singhh** [comment]
Ok!
Please guide me further once you have a look!

**Comment by AskAlexSharov** [comment]
Added hard-coded private key for --chain dev https://github.com/ledgerwatch/erigon/pull/2639
you can remove --miner.sigkey flag

**Comment by 0xKrishna** [comment]
Gives this error now
```krishna@ ~/erigon (devnet_private_key) $ erigon --datadir dev --chain dev --mine
INFO[09-07|14:31:37.967] Build info                               git_branch=krishna/add-init git_tag=v2021.08.01-105-g2e9718599 git_commit=2e9718599556d7280bd3a109c7de061adeea9cf6
INFO[09-07|14:31:37.967] Starting Erigon in ephemeral dev mode...
INFO[09-07|14:31:37.971] Maximum peer count                       ETH=100 total=100
INFO[09-07|14:31:37.971] Set global gas cap                       cap=50000000
Fatal: Please specify developer account address using --miner.etherbase```

**Comment by 0xKrishna** [comment]
When `miner.etherbase` is used
```
krishna@ ~/erigon (devnet_private_key) $ erigon --datadir dev --chain dev --mine --miner.etherbase 0x03f50EE89696c10Ec85Db312535D52fF5868b29A
INFO[09-07|14:32:22.599] Build info                               git_branch=krishna/add-init git_tag=v2021.08.01-105-g2e9718599 git_commit=2e9718599556d7280bd3a109c7de061adeea9cf6
INFO[09-07|14:32:22.599] Starting Erigon in ephemeral dev mode...
INFO[09-07|14:32:22.601] Maximum peer count                       ETH=100 total=100
INFO[09-07|14:32:22.602] Set global gas cap                       cap=50000000
INFO[09-07|14:32:22.602] Using developer account                  address=0x03f50EE89696c10Ec85Db312535D52fF5868b29A
INFO[09-07|14:32:22.603] Opening Database                         label=chaindata path=dev/chaindata
INFO[09-07|14:32:22.614] database closed                          label=chaindata
INFO[09-07|14:32:22.614] Opening Database                         label=chaindata path=/Users/krishna/erigon/dev/chaindata
INFO[09-07|14:32:22.628] Initialised chain configuration          config="{ChainID: 1337 Homestead: 0 DAO: <nil> DAOSupport: false EIP150: 0 EIP155: 0 EIP158: 0 Byzantium: 0 Constantinople: 0 Petersburg: 0 Istanbul: 0, Muir Glacier: 0, Berlin: 0, London: <nil>, Engine: clique}"
INFO[09-07|14:32:22.636] Initialising Ethereum protocol           network=1337
INFO[09-07|14:32:22.637] Effective                                prune="--prune="
INFO[09-07|14:32:22.637] Starting private RPC server              on=127.0.0.1:9090
WARN[09-07|14:32:22.638] Preverified hashes not found for         chain=
INFO[09-07|14:32:22.642] Transaction pool price threshold updated price=0x1
EROR[09-07|14:32:22.642] Etherbase account unavailable locally    err=nil
EROR[09-07|14:32:22.642] Erigon startup
```

**Comment by AskAlexSharov** [comment]
@0xKrishna added default sign key for Devnet: https://github.com/ledgerwatch/erigon/pull/2639
Now must be enough: `--chain dev --mine`

**Comment by 0xKrishna** [comment]
Even that's not working
```
krishna@ ~/erigon (devnet_private_key) $ erigon --datadir dev --chain dev --mine
INFO[09-07|14:36:29.361] Build info                               git_branch=krishna/add-init git_tag=v2021.08.01-105-g2e9718599 git_commit=2e9718599556d7280bd3a109c7de061adeea9cf6
INFO[09-07|14:36:29.362] Starting Erigon in ephemeral dev mode...
INFO[09-07|14:36:29.364] Maximum peer count                       ETH=100 total=100
INFO[09-07|14:36:29.365] Set global gas cap                       cap=50000000
Fatal: Please specify developer account address using --miner.etherbase```

**Comment by AskAlexSharov** [comment]
@0xKrishna are you on https://github.com/ledgerwatch/erigon/pull/2639 ?

**Comment by 0xKrishna** [comment]
Sorry, My bad
```
krishna@ ~/erigon (devnet_private_key) $ ./build/bin/erigon --datadir dev --chain dev --mine```
Working now

**Comment by AskAlexSharov** [comment]
To continue:
@Uttam-Singhh you can see in Erigon logs "local tx propagated" - it means your transaction came to TxPool successfuly and TxPool sent it to other peers by P2P. Now need understand if your transaction came to mined block. 
Mining happening next way:
- `eth/stagedsync/stage_mining_create_block.go` getting txs from Pool 
- `eth/stagedsync/stage_mining_exec` executing transactions and create new block
- `eth/stagedsync/stage_mining_finish.go` sign new block
So, maybe your tx has not enough gas fee or wrong nonce - and miner rejected it by some reason.
Please take a look. Thank you.

**Comment by Uttam-Singhh** [comment]
Oh okay I will try it out!
Thank you so much @AskAlexSharov!❤️

**Comment by AskAlexSharov** [comment]
after https://github.com/ledgerwatch/erigon/pull/2639 you better delete your dev db
and of course now - if you do custom genesis - you need allocate some funds for default etherbase

**Comment by 0xKrishna** [comment]
I have tried everything but my node is not mining blocks at all, Interesting part is this
```
INFO[09-07|17:03:45.136] [1/18 Headers] Wrote block headers       number=0 blk/second=0.000 alloc="10.51 MiB" sys="72.14 MiB"
^CINFO[09-07|17:04:12.572] Got interrupt, shutting down...
INFO[09-07|17:04:12.574] event subscription channel closed with the RPC daemon
INFO[09-07|17:04:12.577] [1/5 MiningCreateBlock] Start mine       block=1
INFO[09-07|17:04:13.579] Transaction pool stopped
```
When I try to stop the node by interrupting, I can see a log related to mining create block

**Comment by 0xKrishna** [comment]
Using similar command in geth works fine
```
 ./build/bin/geth --datadir dev --dev --mine
 
INFO [09-07|18:59:44.165] Commit new mining work                   number=1 sealhash=e891d0..d0aa79 uncles=0 txs=0 gas=0 fees=0 elapsed="143.589µs"
INFO [09-07|18:59:44.165] Sealing paused, waiting for transactions 
```

But in erigon it doesn't mine
```
./build/bin/erigon --datadir dev --chain dev --mine


INFO[09-07|19:19:47.207] [1/18 Headers] Wrote block headers       number=0 blk/second=0.000 alloc="7.33 MiB" sys="72.08 MiB"
INFO[09-07|19:20:17.179] [1/18 Headers] Wrote block headers       number=0 blk/second=0.000 alloc="7.17 MiB" sys="72.08 MiB"
```

**Comment by mmgen** [comment]
> Using similar command in geth works fine
> 
> ```
>  ./build/bin/geth --datadir dev --dev --mine

Is the `--mine` option really required? For Geth devnet, all I need is the following:

    geth --http --http.api=eth,web3,txpool --dev

**Comment by AskAlexSharov** [comment]
I found problem. Mining was blocked on some mutex which didn't happen on active networks(where periodically arriving blocks). Here is some fix: https://github.com/ledgerwatch/erigon/pull/2642

**Comment by 0xKrishna** [comment]
Okay, This is what's happeing now
```
INFO[09-08|11:39:41.333] local tx propagated                      to_peers_amount=0 tx_hash=0xed4eb1564a7d09d04a11fd4998924b1d3b1ddd9a70bd339904c7202b739eddb9
INFO[09-08|11:39:42.908] [1/18 Headers] Processed                 highest inserted=0 age=52y5mo1w
INFO[09-08|11:39:42.909] Commit cycle                             in=40.499µs
INFO[09-08|11:39:42.909] Update current block for the RPC API     from=1 to=0
INFO[09-08|11:39:42.909] [1/5 MiningCreateBlock] Start mine       block=1
INFO[09-08|11:39:42.910] [4/5 IntermediateHashes] Regeneration trie hashes started
INFO[09-08|11:39:42.910] [4/5 IntermediateHashes] Trie root       hash=0xe3eade507848eaf22077585b83e58bd8051d584ef9057c3423ec9aab1a0b7cc8
INFO[09-08|11:39:42.910] [4/5 IntermediateHashes] Regeneration ended
INFO[09-08|11:39:42.910] [5/5 MiningFinish] start
INFO[09-08|11:39:42.910] [5/5 MiningFinish] block ready for seal  blocn_num=1 transactions=2 gas_used=42000 gas_limit=11488771 difficulty=2
INFO[09-08|11:39:42.911] [1/18 Headers] Waiting for headers...    from=0
INFO[09-08|11:39:44.908] [1/18 Headers] Processed                 highest inserted=0 age=52y5mo1w
INFO[09-08|11:39:44.910] Commit cycle                             in=33.766µs
INFO[09-08|11:39:44.910] Update current block for the RPC API     from=1 to=0
INFO[09-08|11:39:44.910] [1/18 Headers] Waiting for headers...    from=0
```
Block is ready for seal but number is till not increasing, Anything else need tp be done ?
Command used to run the chain
```
./build/bin/erigon --datadir dev --chain dev --mine
```

**Comment by AskAlexSharov** [comment]
`but the number is still not increasing, Anything else needs tp be done?`
I think easiest way for now is to start multiple nodes and join them by --staticpeers flag. They will propagate mined block to each-other.  
I will work to apply own mined blocks tomorrow - but I'm not sure - is it easy or hard.

**Comment by 0xKrishna** [comment]
I used `--staticpeers` but how can I make sure they are connected?
Running both the nodes on the same machine by changing the port number from 9090 to 9091. That's fine, Right?
I used RPC of one node to submit the txn, It had all the earlier logs but there were no logs on the 2nd node, Nor the txn got propagated on 2nd node.

**Comment by AskAlexSharov** [comment]
9090 - it's grpc port. you need flag --port 30303

**Comment by AskAlexSharov** [comment]
https://github.com/ledgerwatch/erigon#default-ports-and-protocols--firewalls

**Comment by 0xKrishna** [comment]
Yes, I used this command to run 2nd node
```
 ./build/bin/erigon --datadir dev2 --chain dev --mine --staticpeers "enode://5c754c68a5f1d489c1273c2fc5c4926194e7f6588a90733b74f6f8292aa20a6585b2d22ae112d65d3f5407d0302843e8c3d174970e6d5ebe2b329532bb8f61a5@127.0.0.1:30304?discport=0" --private.api.addr=localhost:9091 --port 30305 --p2p.eth65.port 30306
 ```
 1st node is using ports 9091, 30303 and 30304
 I hope staticpeers is used correctly.

**Comment by AskAlexSharov** [comment]
Looks correct

**Comment by 0xKrishna** [comment]
I am getting this in logs
```
INFO[09-08|18:13:29.290] [p2p] GoodPeers                          eth66=0 eth65=0
```

**Comment by AskAlexSharov** [comment]
Apply mined blocks: https://github.com/ledgerwatch/erigon/pull/2650

**Comment by 0xKrishna** [comment]
Thanks @AskAlexSharov , It's mining blocks now, Header number moving ahead with transactions.

**Comment by 0xKrishna** [comment]
Single node is working fine now and mining blocks as per the transaction, But if I run another local node and also use `--staticpeer` flag, The 2nd node is running independently and not syncing with the first one. Any suggestion?

**Comment by mandrigin** [comment]
@0xKrishna can you give here both full command lines here?

**Comment by 0xKrishna** [comment]
1st Node
```
./build/bin/erigon --datadir dev --chain dev --mine
```
The enode of 1st node
```
INFO[09-13|11:08:13.871] Started P2P networking                   version=65 self="enode://13379656b590c7db0bcdc58b64b6c2b2bbdb0ff5f30cb264e5112be6afce6dd7082a3c65ea05d8c5bbbe8cec6b27396f01964f56c89627960acd856894fb60cf@127.0.0.1:30304?discport=0" name=erigon/v2021.09.3-alpha-4bd8721d/darwin-amd64/go1.17
```

Then to run 2nd node
```
 ./build/bin/erigon --datadir dev2 --chain dev --mine --staticpeers "enode://13379656b590c7db0bcdc58b64b6c2b2bbdb0ff5f30cb264e5112be6afce6dd7082a3c65ea05d8c5bbbe8cec6b27396f01964f56c89627960acd856894fb60cf@127.0.0.1:30304?discport=0" --private.api.addr=localhost:9091 --port 30305 --p2p.eth65.port 30306
 ```

**Comment by AskAlexSharov** [comment]
@mandrigin , hi. I also created ticket about admin_nodeInfo RPC method - then users can get enode by RPC (now can get from logs only): https://github.com/ledgerwatch/erigon/issues/2679

**Comment by 0xKrishna** [comment]
Any update? I am still not able to setup a local dev network with more than 1 node connected.
Single node is working fine but I am not able to connect nodes using staticpeers flag.

**Comment by mandrigin** [comment]
@0xKrishna one more question: do static peers work if you use, say Ethereum mainnet or any testnet?

**Comment by mandrigin** [comment]
We will try to reproduce on our side as well

**Comment by 0xKrishna** [comment]
@mandrigin Haven't tried static peer for mainnet or any testnet, I just use
```
--chain goerli
```
and it works for, Also I don't know any static peer node for goerli or mainnet.

**Comment by 0xKrishna** [comment]
> 1st Node
> 
> ```
> ./build/bin/erigon --datadir dev --chain dev --mine
> ```
> 
> The enode of 1st node
> 
> ```
> INFO[09-13|11:08:13.871] Started P2P networking                   version=65 self="enode://13379656b590c7db0bcdc58b64b6c2b2bbdb0ff5f30cb264e5112be6afce6dd7082a3c65ea05d8c5bbbe8cec6b27396f01964f56c89627960acd856894fb60cf@127.0.0.1:30304?discport=0" name=erigon/v2021.09.3-alpha-4bd8721d/darwin-amd64/go1.17
> ```
> 
> Then to run 2nd node
> 
> ```
>  ./build/bin/erigon --datadir dev2 --chain dev --mine --staticpeers "enode://13379656b590c7db0bcdc58b64b6c2b2bbdb0ff5f30cb264e5112be6afce6dd7082a3c65ea05d8c5bbbe8cec6b27396f01964f56c89627960acd856894fb60cf@127.0.0.1:30304?discport=0" --private.api.addr=localhost:9091 --port 30305 --p2p.eth65.port 30306
> ```

Let me know if you see any issue in these commands which I used for testing

**Comment by Uttam-Singhh** [comment]
Any update?

**Comment by Cript** [comment]
Peers connection for dev network is disabled here: https://github.com/ledgerwatch/erigon/blob/b890e973db294fccff0622da94a324c335f993d8/cmd/utils/flags.go#L848

Output to console after I removed these lines:
```
DBUG[10-01|16:10:32.992] [4ece8d817c5c7f9d324b5cd68778a99cde80627d50cd02b25430194bd540ad32] Received status message OK name=erigon/v2021.10.1/li...
DBUG[10-01|16:10:32.992] [4ece8d817c5c7f9d324b5cd68778a99cde80627d50cd02b25430194bd540ad32] Error while running peer: reading message: EOF 
```
The error happens here: https://github.com/ledgerwatch/erigon/blob/044aa6748cf92e6f9e45ae51f467bf22eb084e1c/p2p/server.go#L992

```
INFO[10-04|16:29:47.140] Dial error                               id=4ece8d817c5c7f9d324b5cd68778a99cde80627d50cd02b25430194bd540ad32 addr=127.0.0.1:30304 conn=staticdial err="useless peer"
```


Metamask works fine with the first node, but the second node is not synced 
It could be an error in my local environment. I am digging further.

**Comment by Cript** [comment]
@AskAlexSharov, is it safe to remove completely the condition here? https://github.com/ledgerwatch/erigon/blob/b890e973db294fccff0622da94a324c335f993d8/cmd/utils/flags.go#L848

**Comment by AskAlexSharov** [comment]
@Cript  , I think no - because Geth also has it. 
But you don't need P2P discovery - you need connect known nodes by --staticpeers flag, right?
--staticpeers must not limited by `cfg.MaxPeers`.

**Comment by AskAlexSharov** [comment]
`ListenAddr = ":0"` - means find some free port. It done this way to create swarm of dev nodes for simulations. 
Geth then can return Enode address by `admin_nodeInfo` RPC call, we don't have it now - but maybe can implement one (now we return Enode address only in logs).

**Comment by 0xKrishna** [comment]
Now not even a single node is working, Earlier when I used to submit txn on local using metamask it used to create block, But now it's not even doing that. Getting this
```
INFO[10-07|18:06:40.396] RPC Daemon notified of new headers       from=0 to=0
INFO[10-07|18:06:40.396] [1/18 Headers] Waiting for headers...    from=0
INFO[10-07|18:06:43.399] [1/18 Headers] Processed                 highest inserted=0 age=52y6mo1w
INFO[10-07|18:06:43.400] Commit cycle                             in=36.732µs
INFO[10-07|18:06:43.400] [1/5 MiningCreateBlock] Start mine       block=1
INFO[10-07|18:06:43.400] RPC Daemon notified of new headers       from=0 to=0
INFO[10-07|18:06:43.401] [4/5 IntermediateHashes] Regeneration trie hashes started
INFO[10-07|18:06:43.401] [4/5 IntermediateHashes] Trie root       hash=0x27a9a4df6cc06489041bf7e5627ef6ac2afb6a14e980436f5afad60179da7cc2
INFO[10-07|18:06:43.401] [4/5 IntermediateHashes] Regeneration ended
INFO[10-07|18:06:43.401] Sealing paused, waiting for transactions
INFO[10-07|18:06:43.401] [1/18 Headers] Waiting for headers...    from=0
INFO[10-07|18:06:46.395] [1/18 Headers] Processed                 highest inserted=0 age=52y6mo1w
INFO[10-07|18:06:46.395] Commit cycle                             in=16.95µs
```

Used the same commands as above to run the node

**Comment by Uttam-Singhh** [comment]
Hello @mandrigin,
I think so that the issue is still unresolved as @0xKrishna commented that Now not even a single node is working, earlier it used to work.
I would be grateful if you can reopen this issue again!😇

**Comment by 0xKrishna** [comment]
I used the same command as above in older chat
```
./build/bin/erigon --datadir dev --chain dev --mine
```
